File: Python/Ejercicio 3/Interfaz_consola.py
def pedir_numero_jugadores():
    return int(input("Ingrese el número de jugadores: "))

def pedir_nombre_jugador(indice):
    return input(f"Jugador {indice}, ingrese su nombre: ")

def preguntar_carta():
    opcion = input("¿Quieres otra carta? (s/n): ").lower()
    return opcion == "s"

def mostrar_mensaje(msg):
    print(msg)

def mostrar_cartas_restantes(num):
    print(f"Cartas restantes en la pila: {num}")

def mostrar_ganador(ganador):
    print(f"¡{ganador['nombre']} gana con {ganador['puntos']} puntos!")

File: Java/Ejercicio 3/InterfazConsola.java
import java.util.Scanner;

public class InterfazConsola {
    private static final Scanner sc = new Scanner(System.in);

    public static int pedirNumeroJugadores() {
        System.out.print("Ingrese el número de jugadores: ");
        return sc.nextInt();
    }

    public static String pedirNombreJugador(int indice) {
        System.out.print("Jugador " + indice + ", ingrese su nombre: ");
        return sc.next();
    }

    public static boolean preguntarCarta() {
        System.out.print("¿Quieres otra carta? (s/n): ");
        String opcion = sc.next().toLowerCase();
        return opcion.equals("s");
    }

    public static void mostrarMensaje(String msg) {
        System.out.println(msg);
    }

    public static void mostrarCartasRestantes(int num) {
        System.out.println("Cartas restantes en la pila: " + num);
    }

    public static void mostrarGanador(String nombre, int puntos) {
        System.out.println("¡" + nombre + " gana con " + puntos + " puntos!");
    }
}
